mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala

```diff
@@ -408,7 +408,7 @@ private[python] class PythonMLLibAPI extends Serializable {
     if (seed != null) gmmAlg.setSeed(seed)
 
     try {
-      new GaussianMixtureModelWrapper(gmmAlg.run(data.rdd.persist(StorageLevel.MEMORY_AND_DISK)))
+      new GaussianMixtureModelWrapper(gmmAlg.run(data.rdd))
```

Comment:
For me, this looks not harmful for the performance. Also, this PR seems to break the symmetric structure of `try` and `finally`. If this has no effective good, I'd like to recommend to avoid this change.

Reply:
I think the point is that the implementation also caches a small transformation of it, so this one is kind of pointless. But yes, then the unpersist() should be removed below.

I'm OK with it but we could fix some issues in GaussianMixture while we're here. It never unpersists the data it caches internally, when it should do so. It should also arguably respect whether the given RDD is already cached, but maybe we skip that here.

Reply:
But I think, it's unnecessary to persist the same rdd twice?
- First time, it is getting persisted here.
- Second time, it is persisted (cached) as below.
https://github.com/apache/spark/blob/710ddab39e20f49e917311c3e27d142b5a2bcc71/mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala#L167-L171

Reply:
It wouldn't matter as much if it were the same RDD, but it's not - it's mapped to breeze vectors. It makes more sense to keep this one inside run(). But it needs to be unpersisted.

Reply:
> Also, this PR seems to break the symmetric structure of try and finally.

This could be corrected. Before this, which operation (persist or cache) should be avoided?